components/script/dom/window.rs

```diff
@@ -1356,7 +1402,9 @@ impl Window {
     }
 
     pub fn get_url(&self) -> Url {
-        (*self.Document().url()).clone()
+        let doc = self.Document();
+        let val = (*doc.url()).clone();
+        val
```

Comment:
Actually, this is very odd, since document.url() returns a `&Url`, but in this function we purposefully deref it just to return a `Url`. I suppose this is the status quo until #13790 is solved.

Regardless of return values, we still need to figure out _why_ exactly the lifetimes aren't working out as expected. It is not normal to see how using a `&Url` causes no borrrowck issues but a `Ref<Url>` with an equivalent lifetime causes a borrow error.